### PR TITLE
ci: improve version update script

### PR DESCRIPTION
**Pull request by yihau**
#### Problem

There are some issues in our `scripts/increment-cargo-version.sh`
1. too slow
2. always generates unrelated changes

#### Summary of Changes

- use d0732927ebca3826044c3d7cf57be2b64f4172d0 to fix/improve the skipping condition

- introduce 3c9c2eb1932fabf2972c8d418f78ee803f575902 for discarding unrelated changes

### CODE REVIEW

File: scripts/increment-cargo-version.sh [hunk 1]
```diff
@@ -146,6 +146,18 @@ done
 # Update cargo lock files
 scripts/cargo-for-all-lock-files.sh tree >/dev/null
 
+# Only apply related changes
+(
+  shopt -s globstar
+  # shellcheck disable=SC2035
+  git diff --unified=0 **/Cargo.lock >cargo-lock-patch
```
Comment (willhickey) [inline scripts/increment-cargo-version.sh]: I think we can remove the shellcheck exceptions by prefacing the globstar with `./`. I tried it locally on v1.18 and the output was identical.

```suggestion
    git diff --unified=0 ./**/Cargo.lock >cargo-lock-patch
```

File: scripts/increment-cargo-version.sh [hunk 2]
```diff
@@ -146,6 +146,18 @@ done
...
+  grep -E '^(diff|index|---|\+\+\+|@@.*@@ name = .*|-version|\+version)' cargo-lock-patch >filtered-cargo-lock-patch
+  # shellcheck disable=SC2035
+  git checkout **/Cargo.lock
```
Comment (willhickey) [inline scripts/increment-cargo-version.sh]:
```suggestion
  git checkout ./**/Cargo.lock
```

**Review by willhickey — APPROVED** [review]
Thanks for fixing this! Approved with or without the suggestion

**Review by willhickey — DISMISSED** [review]
Thanks for fixing this! Approved with or without the suggestion

**Comment by yihau** [comment]
thank you @willhickey! good catch! just fixed both of them in [02fefbd](https://github.com/anza-xyz/agave/pull/1553/commits/02fefbdc6ed2503e270a1292c2c59ae89c8f3e2f)!

**Review by willhickey — APPROVED** [review]
:shipit:

**Pull request merged**